pkg/simpleschema/transform.go

```diff
@@ -15,21 +15,31 @@ package simpleschema
 
 import (
 	"fmt"
+	"slices"
 	"strconv"
 	"strings"
 
 	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 )
 
+// A predefined type is a type that is predefined in the schema.
+// It is used to resolve references in the schema, while capturing the fact
+// whether the type has the required marker set (this information would
+// otherwise be lost in the parsing process).
+type predefinedType struct {
```

Comment:
Good call! initially i wasn't sure about this, but i just remembered that required fields always live parent-type-schema-definition.

Reply:
I just noticed that adding this broke the test for the predefined types. So, I fixed that.